### PR TITLE
WIP Http resource leak test coverage and fix

### LINKED ISSUE #17524: Use assertAllStagesStopped in Http tests
@patriknw created this helper method (in stream.testkit.Utils) which helped to uncover many of the leaks we had in streams-core. We whould use this in Http tests, too (where applicable).

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #17524 and #20298

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala [hunk 1]
```diff
@@ -71,4 +71,13 @@ abstract class HttpServerTestSetupBase {
   def send(string: String): Unit = send(ByteString(string.stripMarginWithNewline("\r\n"), "UTF8"))
 
   def closeNetworkInput(): Unit = netIn.sendComplete()
+
+  def shutdownBlueprint(): Unit = {
+    netIn.sendComplete()
+    requests.expectComplete()
+
+    responses.sendComplete()
+    netOut.expectBytes(ByteString("HTT")) // ???
```
Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: @ktoso Wat?

**Comment by johanandren** [comment]
That was a suspciously quick CI run...

File: akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala (hunk 1 above)
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: Well, looks like beginning of the `HTTP/1.1` response... most likely this should not happen but I have not read the sources in detail here yet

Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: Ok, then at least there was no "Well obviously ...". I will take another look at it before this PR is done.

Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: Arh. Already figured it out, I have used it extensively in tests not consuming the reply, so this is of course the start of the reply... :P sorry for pinging you with such a stupid miss.

Comment (2beaucoup) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: But why does this succeed even if there's no response sent at all?

**Comment by johanandren** [comment]
Just a note to myself about this PR, it is waiting for me to look into the weird 'HTT' response in the shutdown.

File: akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala (hunk 1 above)
Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerTestSetupBase.scala]: I have opened a separate ticket #20377 to look at this so we can get the resource leak fix in. Pretty sure the `HTT` behavior of the server in this case wasn't really introduced here, just noticed so it should be safe to merge.

**Comment by johanandren** [comment]
@ktoso @patriknw review thanks!

**Comment by ktoso** [comment]
LGTM, thanks for opening the other ticket

**Pull request merged**